src/borg/archiver.py

```diff
@@ -2168,6 +2168,13 @@ def main():  # pragma: no cover
     except RemoteRepository.RPCError as e:
         msg = '%s\n%s' % (str(e), sysinfo())
         exit_code = EXIT_ERROR
+    except BrokenPipeError:
+        if args.func.__name__ in ['do_info', 'do_list', 'do_diff']:
+            sys.stderr.close()
+            sys.exit(0)
```

Comment:
well, a flag close_stderr could be set and later stderr be closed (while keeping the --show-rc functionality).